nodeClient.go

```diff
@@ -321,31 +321,8 @@ func (rc *NodeClient) getBlockCommon(restUrl *url.URL, withTransactions bool) (d
 // WaitForTransaction does a long-GET for one transaction and wait for it to complete.
 // Initially poll at 10 Hz for up to 1 second if node replies with 404 (wait for txn to propagate).
 // Accept option arguments PollPeriod and PollTimeout like PollForTransactions.
-func (rc *NodeClient) WaitForTransaction(txnHash string, options ...any) (data *api.Transaction, err error) {
-	period, timeout, err := getTransactionPollOptions(100*time.Millisecond, 1*time.Second, options...)
-	if err != nil {
-		return nil, err
-	}
-	restUrl := rc.baseUrl.JoinPath("transactions/wait_by_hash", txnHash)
-	start := time.Now()
-	deadline := start.Add(timeout)
-	for {
-		data, err = rc.getTransactionCommon(restUrl)
-		if err == nil {
-			return
-		}
-		var httpErr *HttpError
-		if errors.As(err, &httpErr) {
-			if httpErr.StatusCode == 404 {
-				if time.Now().Before(deadline) {
-					time.Sleep(period)
-				} else {
-					return
-				}
-			}
-		}
-	}
-
+func (rc *NodeClient) WaitForTransaction(txnHash string, options ...any) (data *api.UserTransaction, err error) {
+	return rc.PollForTransaction(txnHash, options...)
```

Comment:
API perceived latency will get worse by on average (period/2) by no longer using the `transactions/wait_by_hash` API endpoint

Reply:
That's okay.  I need to come back and fix the behavior on it, this was causing some test variability